src/Root/Root.js

```diff
@@ -1,19 +1,23 @@
-import React, { PropTypes } from 'react';
+import React from 'react';
 import { Provider } from 'react-redux';
-import { Router } from 'react-router';
-
+import { browserHistory as history, Router } from 'react-router';
+import { syncHistoryWithStore } from 'react-router-redux';
+import configureStore from '../store/configureStore';
+import rootSaga from './rootSaga';
 import routes from './routes';
 
+import '../../node_modules/grommet/grommet.min.css';
+
+const store = configureStore({ history });
+
+const syncedHistory = syncHistoryWithStore(history, store);
 
-export default function Root({ store, history }) {
+store.runSaga(rootSaga);
+
+export default function Root() {
   return (
     <Provider store={store}>
-      <Router history={history} routes={routes} />
+      <Router history={syncedHistory} routes={routes} />
     </Provider>
   );
 }
```

Comment:
Interesting, this code is the same as code in my job. Would you like to share your reference if there is one? Thanks.